src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java

```diff
@@ -69,4 +72,24 @@ private void validateDoctor(DoctorDto doctorDto) throws CustomException {
             throw new CustomException(DoctorConstants.AADHAAR_NUMBER_PATTERN_MESSAGE);
         }
     }
+
+    /**
+     * @param doctorId UUID of the doctor.
+     * @return DoctorDto object containing doctor information.
+     * @throws CustomException : Class to handle custom exception
+     */
+    public DoctorDto getDoctorById(UUID doctorId) throws CustomException {
+        Optional<DoctorModel> doctorModelOptional = doctorRepository.findById(doctorId);
+        if (doctorModelOptional.isEmpty()) {
+            throw new CustomException(MessageConstants.DOCTOR_NOT_FOUND);
+        }
```

Comment:
Instead of printing message DOCTOR_NOT_FOUND, 
Print specific message.
In this scenario: For the given ID, the doctor is unavailable.

Reply:
changed to "The doctor is unavailable for the given ID"